runtime/compiler/trj9/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -42,6 +42,97 @@
 #include "codegen/CodeGenerator.hpp"                      // for CodeGenerator
 #include "optimizer/TransformUtil.hpp"
 
+/*
+ * \brief transform the arguments of case conversion methods to make the call node friendly to code generator
+ *
+ * \details
+ *    The original case conversion method call has 2 children: the src string obj and the dst string obj which will be 
+ *    the converted string obj.
+ *    After the transformation, the 2 children are replaced with 3 new children: the src string array address, the dst string 
+ *    array address and the string length. The code generator can then focus on converting the actual content of the array and
+ *    doesn't need to calculate the various offsets on different platforms.
+ *
+ *    Before the transformation:
+ *    icall
+ *       =>aload src string obj
+ *       =>aload dst string obj
+ *
+ *    After the transformation:
+ *    icall
+ *       =>aladd
+ *          =>aloadi String.value 
+ *             =>aload src string obj
+ *          lonst arrayheaderoffset  
+ *       =>aladd
+ *          =>aloadi String.value 
+ *             =>aload dst string obj
+ *          lonst arrayheaderoffset  
+ *       =>aloadi String.count
+ *          =>aload src string obj
+ */
+void J9::RecognizedCallTransformer::processCaseConversion(TR::Node* node)
+   {
+   TR::Node *srcStringObject = node->getFirstChild();
+   TR::Node *dstStringObject = node->getSecondChild();
+
+   /* Offset of java/lang/String.value */
+   uint32_t stringValueFieldOffset = 0;
+   char *valueFieldString = NULL;
+   TR_OpaqueClassBlock *stringClass = comp()->fej9()->getClassFromSignature("Ljava/lang/String;", strlen("Ljava/lang/String;"), comp()->getCurrentMethod(), true);
+
+   if (comp()->fej9()->getInstanceFieldOffset(stringClass, "value", "[B") != ~0)
```

Comment:
wouldn't != -1 be more clear

Reply:
yea, but the return type of getInstanceFieldOffset is unsigned int so we have to use ~0